test/unit/carousel.ts

```diff
@@ -0,0 +1,414 @@
+      const carousel = new Carousel();
+
+      expect(carousel.props.speed).to.eq(0);
+      expect(carousel.props.slidestoscroll).to.eq(1);
+      expect(carousel.props.slidestoshow).to.eq(1);
+    });
+  });
+
+  describe('functions', () => {
+    let carousel: Carousel;
+    beforeEach(() => {
+      carousel = new Carousel();
+      carousel.props.items = [
+        { content: 'test0' },
+        { content: 'test1' },
+        { content: 'test2' },
+        { content: 'test3' },
+        { content: 'test4' },
+        { content: 'test5' },
+        { content: 'test6' },
+        { content: 'test7' },
+        { content: 'test8' },
+        { content: 'test9' },
+      ];
+    });
+
+    describe('onMount()', () => {
+      it('should add event listener for window resize', () => {
+        const addEventListener = spy();
+        stub(utils, 'WINDOW').returns({ addEventListener });
+
+        carousel.onMount();
+
+        expect(addEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('onUpdate()', () => {
+      it('should add cloned items to state', () => {
+        const clone = carousel.cloneItems = spy();
+
+        carousel.onUpdate();
+
+        expect(clone).to.be.called;
+      });
+    });
+
+    describe('onUnmount()', () => {
+      it('should add event listener for window resize', () => {
+        const removeEventListener = spy();
+        stub(utils, 'WINDOW').returns({ removeEventListener });
+
+        carousel.onUnmount();
+
+        expect(removeEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('moveNext', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.moveNext();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(1);
+      });
+    });
+
+    describe('movePrevious', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.movePrevious();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(-1);
+      });
+    });
+
+    describe('isSwipeToNext()', () => {
+      it('should move next when swiping left with an angle between 315 and 360', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, 4);
+        expect(direction).to.be.true;
+      });
+
+      it('should move next when swiping left with an angle between 0 and 45', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, -10);
+        expect(direction).to.be.true;
+      });
+
+      it('should move previous when swiping right with an angle between 135 and 225', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 25, 10);
```

Comment:
either separate the action and expectations with a newline or inline the expectation:
```js
expect(Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 25, 10)).to.be.false;
```